apps/designer/app/shared/db/assets.server.ts

```diff
@@ -15,11 +18,30 @@ export const loadByProject = async (projectId?: Project["id"]) => {
 
 export const create = async (
   projectId: Project["id"],
-  values: { type: string; name: string; path: string }
+  values: { name: string; path: string },
+  absolutePath?: string
 ) => {
+  let size = 0;
+  const image = await sharp(absolutePath);
+  const metadata = await image.metadata();
+
+  // if there is no absolute path then it means it's a remote image and we can use fetch
+  if (absolutePath) {
+    size = fs.statSync(absolutePath).size;
```

Comment:
never use sync calls on a server after the bootstrap phase, that's very important

Reply:
also didn't we receive size from the browser as we were uploading? I am sure we did

Reply:
Fixed!